### ISSUE TITLE
empty_enum_arguments false positive with switch statement

### ISSUE THREAD

**Issue by matheusalano** [realm/SwiftLint#3444]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

A switch statement with multiple arguments where at least one argument in a `case` is unused (`_`) is triggering the empty_enum_arguments. However, when I remove the parens around the underscore, the rule is not triggered. 

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint --quiet
/Users/xxx/iOS/xxx/xxx.swift:30:50: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated values if they are not used. (empty_enum_arguments)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? **0.41.0**
* Installation method used (Homebrew, CocoaPods, building from source, etc)? **Homebrew**
* Paste your configuration file:

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? **yes**
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)? **Xcode 11.7 Build version 11E801a**
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
switch (baz, qux) {
// This triggers a violation:
case let (.custom(foo), (_)):
      self.bar = foo

// This doesn't trigger a violation:
case let (.custom(quz), _):
      self.bar = quz
}
```
